csm/common/ha/cluster_management/operations.py

```diff
@@ -179,3 +181,56 @@ def execute(self, cluster_manager, **kwargs):
             err_msg = f"{const.CLUSTER_OPERATIONS_ERR_MSG} : {e}"
             Log.error(err_msg)
 
+class NodeMarkFailure(Operation):
+    """
+    Process mark node failure request.
+    """
+    def validate_arguments(self, **kwargs):
+        """
+        Validates node_id provided by user.
+        :param kwargs: arguments with node_id
+        :returns:
+        """
+        arguments = kwargs.get('arguments', "")
+        num_node_id = int(Conf.get(const.CSM_GLOBAL_INDEX, const.NUM_NODE_ID))
+        Log.info("Validating node_id")
+        for node_id in range(num_node_id):
+            valid_node_id = Conf.get(const.CSM_GLOBAL_INDEX, f"{const.KEY_NODE_ID}[{node_id}]")
+            if arguments['id'] == valid_node_id:
+                return
+        raise InvalidRequest('Request body is missing or invalid request body.')
+
+    def _create_payload(self, **kwargs):
+        """
+        Create payload for tagging node failure.
+        :param : arguments with node_id
+        :returns:
+        """
+        cluster_id = Conf.get(const.CSM_GLOBAL_INDEX, const.KEY_CLUSTERID)
+        arguments = kwargs.get('arguments', "")
```

Comment:
Keep 'arguments'/'id' in Const file if not present

Reply:
Updated